core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -137,28 +137,78 @@ def __init__(self,
         self.compile_time = None  # Time taken to compile the model in seconds
         self.val_score = None  # Score with eval_metric (Validation data)
 
+        self._user_params, self._user_params_aux = self._init_user_params(params=hyperparameters)
+
         self.params = {}
         self.params_aux = {}
-
-        if hyperparameters is not None:
-            hyperparameters = hyperparameters.copy()
-        if hyperparameters is not None and AG_ARGS_FIT in hyperparameters:
-            self._user_params_aux = hyperparameters.pop(AG_ARGS_FIT)  # TODO: Delete after initialization?
-        else:
-            self._user_params_aux = None
-        if self._user_params_aux is None:
-            self._user_params_aux = dict()
-        self._user_params = hyperparameters  # TODO: Delete after initialization?
-        if self._user_params is None:
-            self._user_params = dict()
-
         self.params_trained = dict()
         self._is_initialized = False
         self._is_fit_metadata_registered = False
         self._fit_metadata = dict()
 
         self._compiler = None
 
+    @classmethod
+    def _init_user_params(cls, params: Optional[Dict[str, Any]], ag_args_fit: str = AG_ARGS_FIT, ag_arg_prefix: str = AG_ARG_PREFIX) -> (Dict[str, Any], Dict[str, Any]):
+        """
+        Given the user-specified hyperparameters, split into `params` and `params_aux`.
+
+        Parameters
+        ----------
+        params : Optional[Dict[str, Any]]
```

Comment:
Shouldn't we set the default to None if this value is optional?

Reply:
that is reasonable, added